[CI] PutIndexTemplateRequestTests.testFromXContent fails reproducibly

I saw at least two instances today:

On 6.x: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java10,nodes=virtual&&linux/80/console

This reproduces on 6.x for me:
```
REPRODUCE WITH: ./gradlew :server:test \
  -Dtests.seed=4077509E4F46A969 \
  -Dtests.class=org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateRequestTests \
  -Dtests.method="testFromXContent" \
  -Dtests.security.manager=true \
  -Dtests.locale=en-CC \
  -Dtests.timezone=US/Pacific-New
```

```
10:35:30    > Throwable #1: java.lang.AssertionError: 
10:35:30    > Expected: <[org.elasticsearch.action.admin.indices.alias.Alias@df79bfbc, org.elasticsearch.action.admin.indices.alias.Alias@6d4376, org.elasticsearch.action.admin.indices.alias.Alias@d8ddcb43]>
10:35:30    >      but: was <[org.elasticsearch.action.admin.indices.alias.Alias@6d4376, org.elasticsearch.action.admin.indices.alias.Alias@d9ea79d7, org.elasticsearch.action.admin.indices.alias.Alias@d8ddcb43]>
10:35:30    > 	at __randomizedtesting.SeedInfo.seed([4077509E4F46A969:611F6586C6E149B3]:0)
10:35:30    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
10:35:30    > 	at org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateRequestTests.assertEqualInstances(PutIndexTemplateRequestTests.java:189)
10:35:30    > 	at org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateRequestTests.assertEqualInstances(PutIndexTemplateRequestTests.java:48)
```

--- a/server/src/main/java/org/elasticsearch/action/admin/indices/alias/Alias.java
+++ b/server/src/main/java/org/elasticsearch/action/admin/indices/alias/Alias.java
@@ -28,7 +28,8 @@ import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Streamable;
 import org.elasticsearch.common.xcontent.ToXContent;
-import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
 import org.elasticsearch.common.xcontent.XContentParser;
@@ -42,7 +43,7 @@ import java.util.Map;
 /**
  * Represents an alias, to be associated with an index
  */
-public class Alias implements Streamable, ToXContentObject {
+public class Alias implements Streamable, ToXContentFragment {
 
     private static final ParseField FILTER = new ParseField("filter");
     private static final ParseField ROUTING = new ParseField("routing");
@@ -246,6 +247,11 @@ public class Alias implements Streamable, ToXContentObject {
 
         builder.endObject();
         return builder;
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(this);
     }
 
     @Override

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequestTests.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequestTests.java
@@ -150,7 +150,9 @@ public class PutIndexTemplateRequestTests extends AbstractXContentTestCase<PutIn
         request.patterns(Arrays.asList(generateRandomStringArray(20, 100, false, false)));
         int numAlias = between(0, 5);
         for (int i = 0; i < numAlias; i++) {
-            Alias alias = new Alias(randomRealisticUnicodeOfLengthBetween(1, 10));
+            // some ASCII or Latin-1 control characters, especially newline, can lead to
+            // problems with yaml parsers, that's why we filter them here (see #30911)
+            Alias alias = new Alias(randomRealisticUnicodeOfLengthBetween(1, 10).replaceAll("\\p{Cc}", ""));
             if (randomBoolean()) {
                 alias.indexRouting(randomRealisticUnicodeOfLengthBetween(1, 10));
             }

